resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java

```diff
@@ -3286,6 +3287,68 @@ public void testSrvWithCnameNotCached() throws Exception {
         }
     }
 
+    @Test
+    public void testCNAMEOnlyTriedOnAddressLookups() throws Exception {
+
+        final AtomicInteger cnameQueries = new AtomicInteger();
+
+        TestDnsServer dnsServer2 = new TestDnsServer(new RecordStore() {
+            @Override
+            public Set<ResourceRecord> getRecords(QuestionRecord questionRecord) {
+                if (questionRecord.getRecordType() == RecordType.CNAME) {
+                    cnameQueries.incrementAndGet();
+                }
+
+                return Collections.emptySet();
+            }
+        });
+
+        dnsServer2.start();
+
+        DnsNameResolver resolver = null;
+        try {
+            resolver = newNonCachedResolver(ResolvedAddressTypes.IPV4_PREFERRED)
+                    .maxQueriesPerResolve(3)
+                    .nameServerProvider(new SingletonDnsServerAddressStreamProvider(dnsServer2.localAddress()))
+                    .build();
+
+            try {
+                resolver.resolveAll(new DefaultDnsQuestion("service.netty.io", SRV)).get();
```

Comment:
should we better also verify the exception that was thrown ? Something like:

```
assertThat(instanceof(ExpectedExceptionType.class, 
    resolver.resolveAll(new DefaultDnsQuestion("service.netty.io", SRV)).sync().cause());
```

This is true for all the queries

Reply:
Thanks, will do.

Reply:
@benevans ping we once ready for another review